### ISSUE TITLE
why don't all buttons follow the same naming conventions?

### ISSUE THREAD

**Issue by borisrorsvort** [basscss/basscss-sass#4]
.btn vs .button ?

why having _button-outline.scss vd  _btn-outline.scss

Should I put these issues in the original repo?

**Comment by jxnblk** [comment]
The `.button` style is from previous versions and has been deprecated in favor of `.btn`. They are both offered here to help with backwards compatibility.

**Comment by borisrorsvort** [comment]
Thanks!  Just maybe the doc should reflect that a bit more :)
Do you plan to convert optional packages to the new standard btn? Do you need any help?

**Comment by borisrorsvort** [comment]
Maybe it would also help to have some deprecated comment message in the partials themselfves

**Comment by jxnblk** [comment]
Yeah, comments in the source CSS files would probably be helpful, as well as an npm deprecated warning for the source modules.

If you want to PR any of the modules with comments and a note at the top of the READMEs, I'd be happy to take a look.

**Comment by borisrorsvort** [comment]
Sure no prob. What your plan about these in the future? Are you gonna do more than just renaming button to btn for the v7 update?

**Comment by borisrorsvort** [comment]
Would be nice to say what they are gonna be replace with in future in the deprecation message

**Comment by borisrorsvort** [comment]
@jxnblk I can do the css note but for npm one you need owner rights (https://docs.npmjs.com/api/deprecate)

**Comment by jxnblk** [comment]
Sorry missed this thread, but just going to follow up to address the points here.
- Plans for the future: try to make v8 functionally complete. That means that v7 is very stable, but there will be some small changes here and there. The most volatile parts are the _color_ modules – mostly forms, tables, etc. I'm not decided on it yet, but the core package may or may not include any color styles at all – meaning they would all be optional.
- It would be nice, but most of the breaking changes are minimal and major versions indicate when they happen. The release notes should describe any changes needed for updating. v7 was available as a beta and an open PR for a while to gather feedback on the proposed changes. I'll likely follow a similar process for v8.
- I'll deprecate the npm modules soon.

Anyway, thanks for pointing some of this stuff out. It's super helpful :sunglasses:

**Comment by borisrorsvort** [comment]
@jxnblk  :+1:  :)

Though from a basscss-sass user point of view say its a little confusing to know if one module is from v7 of below since "basscss.scss" includes every single available module. Honestly I though v7 was still something ongoing since not all modules were updated ^^

Shouldn't it only include the one from that version and have a way to include old modules of needed?

If time is missing, I can definitely help since I'm using Bass and Rebass now at my day job and I'd love to make this go faster ;) Just tell me what to do ;)

**Comment by jxnblk** [comment]
Yeah, it's sort of a time issue for me. The assumption being that if you're using Sass, you'd import into your own file, but maybe reference the `basscss.scss` file for source order and to see what's available.

I think a good way to handle this would be to change how the `build.js` script works and only include the core modules here: https://github.com/basscss/basscss-sass/blob/master/build.js#L36

It would mean changing where the two arrays are concatenated.

**Comment by borisrorsvort** [comment]
Why not having two files: basscss.scss and basscss-optional.scss ?

Also there is a confusion between optional modules and deprecated. I think optionals are some modules that can't be replaced by something from the core but should follow the same conventions. Deprecated are modules that cant be replaced and should not be used as such hence deprecated.

**Comment by jxnblk** [comment]
Yeah, that's a fair point, and I've been considering removing the older ones from optional all together. The word _deprecated_ is tricky, because it's still 100% valid to use the old ones. They're just not included in the core package anymore.

**Comment by borisrorsvort** [comment]
It's just that even if basscss is by essence very modular, I have the impression that most pepole would just include @import 'basscss'; to get the whole lib at once. But then you get all the deprecated ones.
And btn vs button is specially confusing ;) 

For the deprecated one let me try again :p 
If they can be replaced by new markups from core:
- add deprecation warning
- suggest alternative
- dont include in the core

If they can't be replaced by new markups from core:
- update them to follow same conventions
- include as optional package
- or delete if dont make sense anymore

**Comment by jxnblk** [comment]
Yeah sorry if I'm being unclear. I think we're pretty much on the same page.

**Comment by borisrorsvort** [comment]
:+1:  I'll see what I can do :wink:

**Comment by jxnblk** [comment]
Cool. I think one place to start would be to just remove the button modules here https://github.com/basscss/basscss/blob/master/index.js#L31-L35
